test/powershell/engine/Module/ModuleCmdlet.Tests.ps1

```diff
@@ -0,0 +1,518 @@
+    }
+
+    It "Passes the imported module out as a custom object when -AsCustomObject is used" {
+        $mod = Import-Module $modInfo -AsCustomObject
+
+        $mod                              | Should -BeOfType "PSCustomObject"
+        $mod.'Test-FirstModuleFunction'() | Should -BeExactly "TESTSTRING"
+    }
+
+    It "Re-imports modules from file with new members when -Force is used" {
+        $module = Import-Module $modInfo -PassThru
+
+        $module.ExportedFunctions.Count | Should -Be 1
+        $module.ExportedFunctions.Keys  | Should -Contain "Test-FirstModuleFunction"
+        $module.ExportedFunctions.Keys  | Should -Not -Contain "Test-SecondModuleFunction"
+
+        $newModuleContent = (Get-Content $modData.Path | Out-String) + "`nfunction Test-SecondModuleFunction { Write-Output 'SECONDSTRING' }"
+        Set-Content -Force -Path $modData.Path -Value $newModuleContent
+
+        Import-Module $modInfo -Force
+        Test-SecondModuleFunction | Should -BeExactly "SECONDSTRING"
+    }
+
+    It "Re-imports modules from file with new members when -Force is used with -PassThru" -Skip {
+        $module = Import-Module $modInfo -PassThru
+
+        $module.ExportedFunctions.Count | Should -Be 1
+        $module.ExportedFunctions.Keys  | Should -Contain "Test-FirstModuleFunction"
+        $module.ExportedFunctions.Keys  | Should -Not -Contain "Test-SecondModuleFunction"
+
+        $newModuleContent = (Get-Content $modData.Path | Out-String) + "`nfunction Test-SecondModuleFunction { Write-Output 'SECONDSTRING' }"
+        Set-Content -Force -Path $modData.Path -Value $newModuleContent
+
+        $module = Import-Module $modInfo -PassThru -Force
+
+        $module.ExportedFunctions.Count | Should -Be 2
+        $module.ExportedFunctions.Keys  | Should -Contain "Test-FirstModuleFunction"
+        $module.ExportedFunctions.Keys  | Should -Contain "Test-SecondModuleFunction"
+    }
+
+    It "Re-imports modules from file with new members when -Force is used with -AsCustomObject" -Skip {
+        $module = Import-Module $modInfo -PassThru
+
+        $module.ExportedFunctions.Count | Should -Be 1
+        $module.ExportedFunctions.Keys  | Should -Contain "Test-FirstModuleFunction"
+        $module.ExportedFunctions.Keys  | Should -Not -Contain "Test-SecondModuleFunction"
+
+        $newModuleContent = (Get-Content $modData.Path | Out-String) + "`nfunction Test-SecondModuleFunction { Write-Output 'SECONDSTRING' }"
+        Set-Content -Force -Path $modData.Path -Value $newModuleContent
+
+        $module = Import-Module $modInfo -AsCustomObject -Force
+        $module.'Test-SecondModuleFunction'() | Should -BeExactly "SECONDSTRING"
+    }
+}
+
+Describe "Import-Module with nested modules" -Tag "Feature" {
+    BeforeAll {
+        SetupModules
+    }
+
+    AfterAll {
+        TearDownModules
+    }
+
+    It "Resolves submodule classes with 'using module'" {
```

Comment:
I don't think `using module` is the right scenario for testing `import-module -force` doesn't refresh nested modules.
`using module` internally loads the module without `-Force`, so by definition forcing reloading the outer module doesn't refresh the ones pulled in by `using module`.

The right scenario should be nested modules or required modules declared in the module manifest I think.

Reply:
Well my intent here isn't to test `-Force` behaviour, just how `Import-Module` works in any scenarios with nested modules (seeing as how this seems to be a complicated area). The `using module` tests are just there to characterise the behaviour. We could move them to their own describe block though. @alx9r had a really nice breakdown of what tests he thought should be written in the old PR.